scrapy/utils/response.py

```diff
@@ -80,8 +81,8 @@ def open_in_browser(
     body = response.body
     if isinstance(response, HtmlResponse):
         if b'<base' not in body:
-            repl = f'<head><base href="{response.url}">'
-            body = body.replace(b'<head>', to_bytes(repl))
+            repl = f'\\1<base href="{response.url}">'
+            body = re.sub(b"(<head.*?>)", to_bytes(repl), body)
```

Comment:
```suggestion
            body = re.sub(b"(<head(?:>|\s.*?>))", to_bytes(repl), body)
```

Only allows `<head` to be followed by `>` or a space-like character, so that things like `<header>` would not be matched. It would be great to include a test for this scenario.

I also thought of specifying `count=0`, but then if there’s a <head> between HTML comments before the actual one things would fail. This scenario would also be nice to cover in tests.

Reply:
Nice catch, I'll redact tests for these scenarii